packages/backend/src/managers/catalogManager.ts

```diff
@@ -274,4 +286,68 @@ export class CatalogManager extends Publisher<ApplicationCatalog> implements Dis
   private getUserCatalogPath(): string {
     return path.resolve(this.appUserDirectory, USER_CATALOG);
   }
+
+  public filterRecipes(filters: RecipeFilters): FilterRecipesResult {
+    let result = this.getRecipes();
+    for (const [filter, values] of Object.entries(filters)) {
+      switch (filter) {
+        case 'languages': {
+          let res: Recipe[] = [];
+          for (const value of values) {
+            res = [...res, ...result.filter(r => r.languages?.includes(value))];
+          }
+          result = res;
+          break;
+        }
+        case 'tools':
+          result = result.filter(r => values.includes(r.backend ?? ''));
```

Comment:
Why is backend used ?

Reply:
the goal would be to include backend as a tool (and for now, only the backend would be part of tools)

Reply:
So we don't filter against the tools field ?

Reply:
In https://github.com/containers/podman-desktop-extension-ai-lab/pull/2297/files, I added only `languages` and `frameworks` fields to the catalog.

My intention was to use the `backend` field for the `tools` search. I suppose we can adjust when the filter fields are more clearly defined. For now, the only reference to tools in the spec is at https://github.com/containers/podman-desktop-extension-ai-lab/issues/1628#issuecomment-2507766815, which seems related to the backend only.